### PR TITLE
Prove entrfi*

### PR DESCRIPTION
**Pull request by BTernaryTau**
> Transitivity of equinumerosity for finite sets, proved without using the Axiom of Power Sets (unlike ~ entr ).

Also revises [nnfi](https://us.metamath.org/mpeuni/nnfi.html), [ssnnfi](https://us.metamath.org/mpeuni/ssnnfi.html), [enfii](https://us.metamath.org/mpeuni/enfii.html), and [enfi](https://us.metamath.org/mpeuni/enfi.html).

Changes in axiom usage can be found here: https://github.com/metamath/set.mm/commit/6793cb65bc09d1a5544c8edaa704b99ac2b5d105

### CODE REVIEW

File: set.mm
Comment (avekens) [inline set.mm:89489]: Tag should be "(Proof shortened by BTernaryTau, 23-Sep-2024.)" instead of "(Revised by BTernaryTau, 23-Sep-2024.)"

**Review by avekens — APPROVED** [review]

File: set.mm
Comment (benjub) [inline set.mm:89805]: I think nnfiOLD and enfiOLD should be kept as ALT since they are shorter, and working in ZF is more common than working in ZF \ Pow.

The comment of xxxALT begins as a rule like: `$( Alternate proof of ~ xxx `, so in this case, it could be
```
$( Alternate proof of ~ nnfi . Shorter but uses ~ ax-pow .  ...
```

On the other hand, enfiiOLD is a direct consequence of enfiOLD, so I think it can remain OLD, and be removed next year.

File: set.mm
Comment (benjub) [inline set.mm:89764]: Is this one really useful, since you have already proved symmetry ?

**Comment by benjub** [comment]
That's pretty nice ! Is the Ackermann interpretation your next goal ? It may also be useful to have the other theorems about binary products of structures xps*

**Review by icecream17 — APPROVED** [review]

File: set.mm
Comment (jkingdon) [reply set.mm:89805]: I would tend to, in general, think that it is worth having both a short proof and a proof which uses fewer axioms.

In this particular case, the difference is length is small enough that I'm kind of neutral on the subject of what to do.

Comment (BTernaryTau) [reply set.mm:89805]: In general I've been reluctant to create ALT proofs during this process because I don't want to clutter set.mm with too many of them, and because the reordering means they often won't be beside the main proof. These two proofs in particular don't seem all that important to keep to me, but if the above two concerns aren't seen as issues, then I could open another PR to turn some of the previous OLD proofs into ALT proofs. In particular I'm thinking of theorems like unfi and ssfi which I proved using induction. And if there's also consensus that nnfiOLD and enfiOLD should be made into ALT proofs, then I can update this PR to do that.

File: set.mm
Comment (BTernaryTau) [reply set.mm:89764]: I personally feel like transitivity is an important enough result that it is worth having all 3, but yes, you could use entrfil along with ensymfib or entrfi along with enfi, depending on what fits best.

**Comment by BTernaryTau** [comment]
> That's pretty nice ! Is the Ackermann interpretation your next goal ? It may also be useful to have the other theorems about binary products of structures xps*

I think that might be beyond my mathematical capabilities at the moment, lol. Honestly this whole thing was motivated by me noticing that the vast majority of theorems about words depend on both ax-pow and ax-rep even though they shouldn't need to and wanting to change that.

File: set.mm
Comment (icecream17) [reply set.mm:89805]: If the ALT proofs are similar in some way, then there could just be one ALT proof standing in for all of them, like in https://us.metamath.org/mpeuni/2cnALT.html

**Comment by benjub** [comment]
> > That's pretty nice ! Is the Ackermann interpretation your next goal ? It may also be useful to have the other theorems about binary products of structures xps*
> 
> I think that might be beyond my mathematical capabilities at the moment, lol. Honestly this whole thing was motivated by me noticing that the vast majority of theorems about words depend on both ax-pow and ax-rep even though they shouldn't need to and wanting to change that.

Actually, I should think about the Ackermann interpretation. There may be a simple argument why one cannot prove any/both directions without these axioms... but if one can, then proving it in set.mm would be an achievement, if you're up for the challenge !  Have a look at https://doi.org/10.1305/ndjfl/1193667707 if you are interested (ignoring the details and the more difficult parts, which are not needed for the present task).

Anyway: what has been proved is already great !

**Comment by langgerard** [comment]
Let TC '("Transitive Containment") be the assertion that for every set x there exits a set u that is the transitive closure of x.
The theorem tz9.1 in set.mm proves TC, using the axiom ax-inf of infinity. the comment of tz9.1 writes that an article of Mancini Antonella and Zambella Domenico seems to prove that ax-inf is ncessary to prove TC.
The reference given by BJ is an article of Kayes Richard and Wong whose thorem 13 gives a nice formulation  and a clear sketch of proof of the result alluded in Mncini and Zambella, as follows:
Let ZF* be the set theory obtained by ommittingt the axiom ax-inf from ZF, then there exists a model of ZF* that proves the negation of TC. 
So that ax-inf is indeed necessary to prove TC over ZF*, and if FinZF  ("the finite ZF theory) is ZF* with the negation of ax-inf added, then FinZF cannot prove TC
I suggest hat the comment in tz9.1 should be modified to become more affirmative that ax-infis needed to have TC.

**Review by jkingdon — APPROVED** [review]
Has an approval, some positive comments, and some suggestions which as far as I can tell don't need to block merging.

If people want to consider those topics further, I guess making a github issue or follow up pull request would be the next way to proceed.

**Pull request merged**
